### PR TITLE
Implement Code and Boolean Stratification

### PR DESCRIPTION
**Pull request by bastianschaffer**
Fixes: #5, #9

Also fixed query parameter bug in DataStore and removed deprecated 'basis' field from all measures.

### CODE REVIEW

File: src/main/java/de/medizininformatikinitiative/fhir_data_evaluator/ComponentExpression.java [hunk 1]
```diff
@@ -61,14 +61,37 @@ public StratumComponent evaluate(Resource resource) {
...
 
-        if (!(found.get(0) instanceof Coding coding)) {
-            return StratumComponent.ofFailedInvalidType(code);
+        if (found.get(0) instanceof Coding coding) {
+            return evaluateCoding(coding);
         }
 
-        if (!coding.hasSystem() || !coding.hasCode()) {
-            return StratumComponent.ofFailedMissingFields(code);
+        if (found.get(0) instanceof Enumeration<?> valueCode) {
```
[the remark this answers is not in the document]
Comment (bastianschaffer) [reply src/main/java/de/medizininformatikinitiative/fhir_data_evaluator/ComponentExpression.java:68]: This is the class implementing the primitive type "code" see: https://hapifhir.io/hapi-fhir/apidocs/hapi-fhir-structures-r4/org/hl7/fhir/r4/model/Enumeration.html

But i forgot to add CodeType, which is also implementing the type "code": https://hapifhir.io/hapi-fhir/apidocs/hapi-fhir-structures-r4/org/hl7/fhir/r4/model/CodeType.html

[the remark this answers is not in the document]
Comment (alexanderkiel) [reply src/main/java/de/medizininformatikinitiative/fhir_data_evaluator/ComponentExpression.java]: Ok I didn't see the package of `Enumeration` and was thinking about `java.util.Enumeration`. But is the HAPI `Enumeration` really used? Is that code path covered by your tests?

File: Documentation/example-measures/example-measure-10.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            "coding": [
+              {
+                "system": "http://fhir-evaluator/strat/system",
+                "code": "observation-value-code"
+              }
+            ]
+          },
+          "criteria": {
+            "language": "text/fhirpath",
+            "expression": "Observation.value.code"
```
[the remark this answers is not in the document]
Comment (bastianschaffer) [reply Documentation/example-measures/example-measure-10.json:48]: yes this works with `Observation.value.code` and also with `(Observation.value as Quantity).code`

File: .github/integration-test/evaluate-exists.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash -e
```
[the remark this answers is not in the document]
Comment (bastianschaffer) [reply .github/integration-test/evaluate-exists.sh:1]: actually the only difference is the output dir. So I could fuse this into one script and set the output dir as an argument.

File: src/main/java/de/medizininformatikinitiative/fhir_data_evaluator/ComponentExpression.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (bastianschaffer) [reply src/main/java/de/medizininformatikinitiative/fhir_data_evaluator/ComponentExpression.java]: Yes, `Enumeration` is used by hapi. For example, in a [FHIR Quantity](https://www.hl7.org/fhir/datatypes.html#Quantity) the `code` and `comparator` fields are both of type `code`, but in hapi a [Quanitiy ](https://hapifhir.io/hapi-fhir/apidocs/hapi-fhir-structures-r4/org/hl7/fhir/r4/model/Quantity.html)one is an Enumeration and the other is a CodeType.
I have not covered both paths in a test though. I will add that.

File: src/test/resources/FhirDataEvaluatorTest/Measures/measure-3.json [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                "system": "http://fhir-evaluator/strat/system",
+                "code": "observation-value-code"
+              }
+            ]
+          },
+          "component": [
+            {
+              "criteria": {
+                "language": "text/fhirpath",
+                "expression": "Observation.value.code"
```
[the remark this answers is not in the document]
Comment (bastianschaffer) [reply src/test/resources/FhirDataEvaluatorTest/Measures/measure-3.json:35]: yes it works. What do you mean by "we had that measure already"? There is also an obervation measure in the github integration test and one in the example measures, but i wouldn't use an example measure or a measure of the github integration test for the java integration test.

File: Documentation/example-measures/example-measure-10.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (alexanderkiel) [reply Documentation/example-measures/example-measure-10.json:48]: ok

File: src/test/resources/FhirDataEvaluatorTest/Measures/measure-3.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (alexanderkiel) [reply src/test/resources/FhirDataEvaluatorTest/Measures/measure-3.json]: ok

**Review by alexanderkiel — APPROVED** [review]

**Pull request merged**